File: mobile/lib/entities/ReportComment.dart

import 'package:Playground/entities/Comment.dart';
import 'package:Playground/entities/User.dart';

class ReportComment {

  int id;
  Comment comment;
  User author;
  String description;

  ReportComment({this.id, this.comment, this.author, this.description});

  factory ReportComment.getDefault() {
    return new ReportComment(
      id: 0,
      description: "",
      comment: null,
      author: null
    );
  }

  Map<String, dynamic> toJson() {
    return <String, dynamic> {
      'id' : this.id,
      'description' : this.description,
      'author' : this.author.toJson(),
      'comment' : this.comment.toJson()
    };
  }

}

File: spring/src/main/java/com/playground/model/entity/ReportComment.java
package com.playground.model.entity;

import lombok.Getter;
import lombok.Setter;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import javax.persistence.*;

@Entity
@Getter
@Setter
@Table(name = "report_comment")
@EntityListeners(AuditingEntityListener.class)
public class ReportComment {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private int id;

    @ManyToOne
    private Comment comment;

    @ManyToOne
    private User author;

    private String description;

    public ReportComment(User author, Comment comment, String description) {
        this.author = author;
        this.comment = comment;
        this.description = description;
    }

    public ReportComment() {
    }
}
